### PR TITLE
Login Page and some Notifications and Strings

### PR DESCRIPTION
**Pull request by Caleb-Nieto**
### Feature / Bug Description
Login Page Added with Storage data

### Solution
Login Page added, authService and apiSlice implemented, store.ts completed, Some Notifications and Strings were added.
Loading spinner indicator was added on Log in button

### Notes
Troubles with gitHub so I had to clone again

### Tickets
[WMA-2](https://cdentalcaregroup.atlassian.net/browse/WMA-2)

### Screenshots / Screencasts
Login Page
image

A notification example
image

Inputs validations
image

[WMA-2]: https://cdentalcaregroup.atlassian.net/browse/WMA-2?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,18 +2,24 @@ import { ConfigProvider } from "antd";
...
+import LoginPage from "./pages/auth/Login";
 
 function App() {
   return (
     <ConfigProvider
       theme={{
         token: {
           colorPrimary: "#e73773",
+          colorLinkHover: "#e73773",
+          colorLinkActive: "#e73773",
```
Comment (imadiaz) [inline src/App.tsx:14]: Just a question: if i change these colors, in the login card, they should be updated right?

File: src/core/authReducer.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import User from "../data/user/user";
+
+interface AuthState {
```
Comment (imadiaz) [inline src/core/authReducer.ts:4]: 🚀 !!

File: src/pages/auth/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import { useAppDispatch } from "../../core/store";
+import { setCredentials } from "../../core/authReducer";
+import { useNavigate } from "react-router-dom";
+import { useSessionStorage } from "../../core/useSessionStorage";
+import User from "../../data/user/user";
+import Strings from "../../utils/Strings";
+import { handleErrorNotification } from "../../utils/Notifications";
+
+const LoginPage = () => {
+  const [passwordVisible, setPasswordVisible] = React.useState(false);
```
Comment (imadiaz) [inline src/pages/auth/Login.tsx:15]: hint: here you can use a better name maybe like "isPasswordVisible" or "shouldShowPassword"

File: src/pages/auth/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  const [login, { isLoading }] = useLoginMutation();
+  const distpatch = useAppDispatch();
+  const navigate = useNavigate();
+  const [getSessionUser, setSessionUser] = useSessionStorage<User>("");
```
Comment (imadiaz) [inline src/pages/auth/Login.tsx:19]: can we use a EMPTY constant from the strings file here?

File: src/pages/auth/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    } catch (error) {
+      handleErrorNotification(error)
+    }
+  };
+
+  return (
+    <div className="flex bg-slate-200 justify-center items-center min-h-screen">
+      <div className="p-6 relative  md:w-96  shadow-2xl bg-slate-900 rounded-2xl">
+        <h1 className="text-center text-3xl block font-semibold text-white">
+          Welcome!
```
Comment (imadiaz) [inline src/pages/auth/Login.tsx:46]: can we use string constants here?

File: src/services/authService.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import User from "../data/user/user";
+import { LoginRequest } from "../data/user/user.request";
+import { apiSlice } from "./apiSlice";
+
+export const authService = apiSlice.injectEndpoints({
```
Comment (imadiaz) [inline src/services/authService.ts:5]: 🚀 !! awesome!

File: src/App.tsx (hunk 1 above)
Comment (Caleb-Nieto) [reply src/App.tsx:14]: No, but let me see something

File: src/pages/auth/Login.tsx (hunk 3 above)
Comment (Caleb-Nieto) [reply src/pages/auth/Login.tsx:15]: Got it

File: src/pages/auth/Login.tsx (hunk 4 above)
Comment (Caleb-Nieto) [reply src/pages/auth/Login.tsx:19]: Okay

File: src/pages/auth/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+  const onFinish = async (values: any) => {
+    try {
+      const data = await login(
+        new LoginRequest(values.email, values.password)
+      ).unwrap();
+
+      setSessionUser(data);
+      distpatch(setCredentials({ ...data }));
+      navigate("/page1");
```
[the remark this answers is not in the document]
Comment (Caleb-Nieto) [reply src/pages/auth/Login.tsx:36]: ooohhh, right!

**Review by imadiaz — APPROVED** [review]

**Pull request merged**